logging/log_processing/json_logging.py

```diff
@@ -0,0 +1,148 @@
+    default_msec_format = "%.19s.%03dZ"
+
+    attribute_mapping = {
+        # LogRecord attributes for which we want new names:
+        "filename": "source:filename",
+        "funcName": "source.function",
+        "levelname": "log_level",
+        "levelno": "log_severity",
+        "lineno": "source.line_number",
+        "module": "source.module",
+        "name": "logger",
+        "pathname": "source.pathname",
+        "process": "process.id",
+        "processName": "process.name",
+        "threadName": "thread.name",
+        # Common context attributes which we want to rename:
+        "function_name": "lambda.function_name",
+        "function_version": "lambda.function_version",
+        "log_stream_name": "cwl.log_stream_name",
+        # LogRecord attributes which we want to suppress:
+        "args": None,
+        "created": None,
+        "msecs": None,
+        "msg": None,
+        "relativeCreated": None,
+        "thread": None,
+    }
+
+    def format(self, record):
+        """Format log record by creating a JSON-format in a string."""
+        data = {}
+        for attr, value in record.__dict__.items():
+            if value is None:
+                continue
+            if attr in self.attribute_mapping:
+                new_name = self.attribute_mapping[attr]
+                if new_name is not None:
+                    data[new_name] = value
+            else:
+                data[attr] = value
+        # The "message" is added last so an accidentally specified message in the extra kwargs is ignored.
+        data["message"] = record.getMessage()
+        # Finally, always add a timestamp.
+        data["gmtime"] = self.formatTime(record)
+        return json.dumps(data, separators=(",", ":"), sort_keys=True)
+
+
+# We don't create the config dict until here so that we can use the classes (instead of class names in strings).
+LOGGING_STREAM_CONFIG = {
+    "version": 1,
+    "disable_existing_loggers": False,
+    "formatters": {"json_formatter": {"()": JsonFormatter}},
+    "filters": {"context_filter": {"()": ContextFilter}},
+    "handlers": {
+        "console": {
+            "class": "logging.StreamHandler",
+            "level": "DEBUG",
+            "formatter": "json_formatter",
+            "filters": ["context_filter"],
+            "stream": "ext://sys.stdout",
+        }
+    },
+    "root": {"level": "INFO", "handlers": ["console"]},
+    "loggers": {"botocore": {"qualname": "botocore", "handlers": ["console"], "level": "WARNING", "propagate": 0}},
+}
+
+
+def configure_logging():
+    logging.config.dictConfig(LOGGING_STREAM_CONFIG)
+
+
+# Just for convenience (avoid having too many imports of logging):
+def getLogger(name):
+    return logging.getLogger(name)
+
+
+def update_context(**kwargs):
+    ContextFilter.update_context(**kwargs)
+
+
+if __name__ == "__main__":
+    configure_logging()
+    update_context(aws_request_id="62E538E9-E9C5-415A-9771-6588F9A1A708")
```

Comment:
love this - the attribute is persistent and other modules/classes will log this attribute